### ISSUE TITLE
(feat): specify device for worker, not scheduler

### ISSUE THREAD

**Issue by nkemnitz** [seung-lab/corgie#3]
Helpful to fully utilize GPU and any remaining available CPU on a node.

**Comment by supersergiy** [comment]
It's a little tricky because each task (model) is running code which is completely custom to that model. Because of this, it is hard to enforce anything about the task on the worker side without constraining the models (for example enforce all models to be pytorch). Do you have any suggestion for a clean solutions to this?

**Comment by nkemnitz** [comment]
So far I am only thinking about the hardcoded tasks within Corgie, such as [RenderTask](https://github.com/seung-lab/corgie/blob/main/corgie/cli/render.py#L90).

For code outside of our control, as with whatever is passed to ApplyModel, I would expect the model to simply raise an Exception in the worker is called with an unsupported backend, or maybe fall back to whatever the worker hardware and the model supports.

**Comment by nkemnitz** [comment]
Maybe rephrasing the issue:
I don't necessarily want to enforce a specific backend, but instead let the workers make the best decision. E.g. pick `cuda` if available and supported by the module, pick `cpu` if supported by the module, else raise Exception.
Right now if I specify `cpu` or no device during scheduling, all workers, including GPU workers will continue to use CPU only, and if I specify `cuda` during scheduling, all CPU workers will crash, even though the module would support cpu.

**Comment by supersergiy** [comment]
Got it. I can think of 2 approaches that would make it general for all models:
1.  Make the processor responsible for using the correct device. This would apply both to custom models and built in processors. Right now, all data in corgie is passed as `pytorch` tensors. We could decide that it is responsibility of the processor to use the device of the tensor passed into it as input. That way, even if the model uses `tensorflow` as backend, it would be responsible for checking the device of the input tensors and doing the computation accordingly. `--device` parameter would be added on the worker script, and it would determine the type of tensors passed around on the worker side.
2. Add arbitrary parameters that can be passed from the worker to the tasks it's executing. `--task_params "{'device': 'cuda'}"` would be added to the worker script. The parameters would be passed on to the `forward(..)` call of each task, the task would pass it on to the custom models (if any). 

I think solution 1. is a little more clean, but solution 2. is more general and might be useful in the future. 

What do you think?
